ocaml/xapi/bios_strings.ml

```diff
@@ -32,26 +78,49 @@ let get_bios_string name =
     else str
   with _ -> ""
 
+let get_baseboard_strings () =
+  let keys = ["baseboard-manufacturer"; "baseboard-product-name";
+              "baseboard-version"; "baseboard-serial-number"] in
+
+  let output = get_output_of_type "2" in
+  let records = match Angstrom.parse_string P.record_list output with
+  | Ok records -> records
+  | Error msg -> []
+  in
+  let values = match List.length records = 0 with
+  | true -> (List.hd records).values
+  | false -> []
+  in
+
+  let convert (key, value) =
+    let key = key
+      |> String.lowercase_ascii
+      |> Str.global_replace (Str.regexp " ") "-"
+      |> Printf.sprintf "baseboard-%s" in
+    let value = match value with
+      | "Not Specified" -> ""
+      | v -> remove_invisible v in
+    key, value in
+
+  List.map convert values
+  |> List.filter (fun (k, v) -> List.mem k keys)
+
 (* Obtain the Type 11 OEM strings from dmidecode, and prepend with the standard ones. *)
 let get_oem_strings () =
   let standard = Xapi_globs.standard_type11_strings in
-  try
-    let result, _ = Forkhelpers.execute_command_get_output dmidecode_prog [dmidecode_prog; "-t11"; "-q"] in
-    let n = List.length standard in
-    let rec loop index a =
-      try
-        let b = String.index_from result a ':' in
-        let c = String.index_from result b '\n' in
-        let str = "oem-" ^ (string_of_int index) in
-        let value = String.trim (remove_invisible (String.sub result (b+2) (c-b-2))) in
-        if value <> "" then
-          (str, value) :: loop (index+1) c
-        else
-          loop index c
-      with _ -> []
-    in
-    standard @ (loop (n+1) 0)
-  with _ -> standard
+  let output = get_output_of_type "11" in
+
+  let records = match Angstrom.parse_string P.record_list output with
+  | Ok records -> records
+  | Error msg -> []
+  in
+  let values = match List.length records = 0 with
+  | true -> (List.hd records).values
+  | false -> []
+  in
+
+  let convert i (_, value) = Printf.sprintf "oem-%d" i, remove_invisible value in
+  standard @ List.mapi convert values
```

Comment:
I think the numbering should start from (List.length standard) + 1, like in the old code, otherwise the oem-X numbers will be duplicated